lightkurve/seismology/core.py

```diff
@@ -210,17 +213,25 @@ def plot_echelle(self, deltanu=None, numax=None,
         ff = freq[int(fmin/fs)-x0:int(fmax/fs)-x0] # Selected frequency range
         pp = power[int(fmin/fs)-x0:int(fmax/fs)-x0] # Power range
 
-        n_rows = int((ff[-1]-ff[0])/deltanu) # Number of stacks to use
-        n_columns = int(deltanu/fs)          # Number of elements in each stack
-
         # Reshape the power into n_rows of n_columns
-        ep = np.reshape(pp[:(n_rows*n_columns)], (n_rows, n_columns))
+        #  Must start and end where the modulus approximately equals zero,
+        #  Otherwise you get a phase shift from the correct (x,y) extents
+        # This implementation will truncate any partially-filled rows.
+
+        mod_zeros = find_peaks( -1.0*(ff % deltanu) )[0]
+        start, approx_end = mod_zeros[0], mod_zeros[-1]
+        approx_samples = (approx_end - start)
+        n_rows = len(mod_zeros) - 1
+        n_columns =  approx_samples // n_rows
+        end = start + n_columns*n_rows
+
+        ep = np.reshape(pp[start : end], (n_rows, n_columns))
```

Comment:
Can you add a few more lines of comments to explain what is happening here?  I don't understand!